doc/aaa/TACACS+ Server Monitor.md

```diff
@@ -0,0 +1,80 @@
+# TACACS+ server monitor design
+
+## Overview
+
+SONiC device usually configured with multiple TACACS+ server, when a server is unreachable, SONiC device will try to connect with next TACACS+ server.
+
+SONiC device will communicate with TACACS+ server in following scenarios:
+1. Remote user login to SONiC device.
+2. Remote user run commands on SONiC device.
+
+There is a timeout for each server, the default value is 5 seconds, this means if the first server not reachable, SONiC device will stuck there when user login or running commands.
+
+To improve this issue, SONiC will add a TACACS+ server monitor to change server priority, a server unreachable or slow response will be downgrade.
+
+### Functional Requirement
+- Monit TACACS+ server unreachable event from syslog.
+- Monit TACACS+ server slow response event from syslog.
+- Change server priority based unreachable event and slow response event.
+- Not change any other server attribute.
+- Not change any other TACACS+ config.
+
+### Syslog format
+- TACACS+ server unreachable event format:
+```
+    failed to connect TACACS+ server [ip address]
+```
+
+- TACACS+ server slow response event format:
+```
+    connect TACACS+ server [ip address] take [time span] ms
+```
+
+# 3 Limitation
+
+- Service priority change will have 1 minutes delay, this is because monit service will run profile every 1 minutes.
+
+# 4 Design
+
+```
+       +------------+ 
+       |    Monit   | 
+       +-----+------+  
+             |          
++------------v--------------+       +---------------------+
+|                           |       |                     |
+|                           |       |                     |
+|     TACACS+ Monitor       |------>|       syslog        |
+|                           |       |                     |
+|                           |       |                     |
++------------+--------------+       +---------------------
+             |                                
+   +---------v---------+               +-------+--------+
+   |                   |               |                |
+   | TACACS config cli +--------------->    ConifgDB    |
```

Comment:
>config cli [](http://example.com/codeflow?start=12&length=10)

ConfigDB is by design user config, and we do not expect daemons inside sonic box change them (there are some exception in current implementation, but we do not want to add more). I agree with @a-barboza "adding some state information to the STATE_DB (COUNTER_DB, or appropriate DB)?"

Reply:
Fixed, new design will generate TACACS config by both CONFIG_DB and COUNTER_DB, and keep CONFIG_DB no change.